src/Components/Filters/Filters.jsx

```diff
@@ -0,0 +1,164 @@
+import { useFilter } from "../../Helper";
+import "./Filters.css";
+import { RatingFilter } from "./RatingFilter";
+
+const Filters = () => {
+  const {
+    filterState: {
+      filters: { sort, priceRange, categories, inStock },
+    },
+    filterDispatch,
+  } = useFilter();
+
+  const clearFilterHandler = () => {
+    filterDispatch({ type: "CLEAR_FILTERS" });
+  };
+
+  const priceHandler = (e) => {
+    filterDispatch({
+      type: "FILTER_BY_PRICE_RANGE",
+      payload: e.target.value,
+    });
+  };
+
+  const highToLowHandler = () => {
+    filterDispatch({ type: "SORT_BY_PRICE", payload: "HIGH_TO_LOW" });
+  };
+
+  const lowToHighHandler = () => {
+    filterDispatch({ type: "SORT_BY_PRICE", payload: "LOW_TO_HIGH" });
+  };
+
+  const availabilityHandler = () => filterDispatch({ type: "FILTER_BY_STOCK" });
+
+  return (
+    <aside className="aside">
+      <div className="filter flex_align__center">
+        <h3 className="h3 txt_bold gray_title">Filters</h3>
+        <button className="btn btn_outline btn_sm" onClick={clearFilterHandler}>
+          Clear
+        </button>
+      </div>
+
+      <div className="price">
+        <h4 className="h4 gray_title">Price</h4>
+        <input
+          type="range"
+          className="range_slider"
+          step="10000"
+          min="0"
+          max="100000"
+          value={priceRange}
+          onChange={priceHandler}
+        />
+        <div>{`INR 0-${priceRange}`}</div>
+      </div>
+
+      <div className="category">
+        <h4 className="h4 gray_title mb_4">Categories</h4>
+        <div className="category_item">
+          <input
+            type="checkbox"
+            name="category"
+            id="earphones"
+            onChange={() =>
+              filterDispatch({ type: "CATEGORY", payload: "ear phones" })
```

Comment:
It is a good practice if you keep you payload name in CAPS with underscore in between,

Reply:
will take care of it.